libs/ebpfnetsh/util.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING
+#include <codecvt>
+#include <ws2def.h>
+#include <ws2ipdef.h>
+// ws2def.h and ws2ipdef.h should be included prior to iphlpapi.h.
+#include <iphlpapi.h>
+#include "ebpf_util.h"
+#include "util.h"
+
+std::string
+down_cast_from_wstring(const std::wstring& wide_string)
+{
+    std::wstring_convert<std::codecvt_utf8<wchar_t>, wchar_t> converter;
+    return converter.to_bytes(wide_string);
+}
+
+ebpf_result_t
+parse_ifindex(_In_z_ const wchar_t* arg, _Out_ uint32_t* if_index)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+    NET_LUID if_luid;
+    NET_IFINDEX local_if_index = 0;
+    wchar_t* end_ptr;
+    uint32_t error = ERROR_SUCCESS;
+
+    *if_index = 0;
+
+    // Check if the input string is an interface index.
+    local_if_index = wcstoul(arg, &end_ptr, 10);
+    if ((local_if_index >= 0) && (*end_ptr == L'\0')) {
+        if (local_if_index != 0) {
+            MIB_IPINTERFACE_ROW row = {};
+            InitializeIpInterfaceEntry(&row);
+            row.InterfaceIndex = local_if_index;
+            // Try IPv4 first.
+            row.Family = AF_INET;
+            error = GetIpInterfaceEntry(&row);
+            if (error != ERROR_SUCCESS) {
+                // Try again with IPv6.
+                row.Family = AF_INET6;
+                error = GetIpInterfaceEntry(&row);
+            }
+        }
+    } else {
```

Comment:
```suggestion
    if ((local_if_index <= 0) || (*end_ptr != L'\0')) {
```
For consistency with other netsh contexts, always treat a positive integer as an ifindex.  Trying to do otherwise would cause confusion due to inconsistency.

Reply:
Changed code to allow +ve integers only, although I had interpreted if_index = 0 as wildcard. But absence of the parameter can mean the same.
I still need to call GetIpInterfceEntry though.

Reply:
I don't follow.  Why do you need to call GetIpInterfaceEntry?  Other contexts don't.

Reply:
The check was to see if the if_index is valid. But I will move that check to net_ebpf_ext. Otherwise all UM surfaces, netsh, bpftool and libbpf need to replicate the above check.